src/lib/datepicker/datepicker.spec.ts

```diff
@@ -95,14 +129,159 @@ describe('MdDatepicker', () => {
       expect(() => testComponent.datepicker.openStandardUi()).toThrow();
     });
   });
+
+  describe('datepicker with startAt', () => {
+    let fixture: ComponentFixture<DatepickerWithStartAt>;
+    let testComponent: DatepickerWithStartAt;
+
+    beforeEach(() => {
+      fixture = TestBed.createComponent(DatepickerWithStartAt);
+      fixture.detectChanges();
+
+      testComponent = fixture.componentInstance;
+    });
+
+    it('explicit startAt should override input value', () => {
+      expect(testComponent.datepicker.startAt).toEqual(new SimpleDate(2010, 0, 1));
+    });
+  });
+
+  describe('datepicker with ngModel', () => {
+    let fixture: ComponentFixture<DatepickerWithNgModel>;
+    let testComponent: DatepickerWithNgModel;
+
+    beforeEach(fakeAsync(() => {
+      fixture = TestBed.createComponent(DatepickerWithNgModel);
+      detectModelChanges(fixture);
+
+      testComponent = fixture.componentInstance;
+    }));
+
+    it('should update datepicker when model changes', fakeAsync(() => {
+      expect(testComponent.datepickerInput.value).toBeNull();
+      expect(testComponent.datepicker.selected).toBeNull();
+
+      let selected = new SimpleDate(2017, 0, 1);
+      testComponent.selected = selected;
+      detectModelChanges(fixture);
+
+      expect(testComponent.datepickerInput.value).toEqual(selected);
+      expect(testComponent.datepicker.selected).toEqual(selected);
+    }));
+
+    it('should update model when date is selected', fakeAsync(() => {
+      expect(testComponent.selected).toBeNull();
+      expect(testComponent.datepickerInput.value).toBeNull();
+
+      let selected = new SimpleDate(2017, 0, 1);
+      testComponent.datepicker.selected = selected;
+      detectModelChanges(fixture);
+
+      expect(testComponent.selected).toEqual(selected);
+      expect(testComponent.datepickerInput.value).toEqual(selected);
+    }));
+
+    it('should mark input dirty after input event', () => {
+      let inputEl = fixture.debugElement.query(By.css('input')).nativeElement;
+
+      expect(inputEl.classList).toContain('ng-pristine');
+
+      dispatchFakeEvent(inputEl, 'input');
+      fixture.detectChanges();
+
+      expect(inputEl.classList).toContain('ng-dirty');
+    });
+
+    it('should mark input dirty after date selected', fakeAsync(() => {
+      let inputEl = fixture.debugElement.query(By.css('input')).nativeElement;
+
+      expect(inputEl.classList).toContain('ng-pristine');
+
+      testComponent.datepicker.selected = new SimpleDate(2017, 0, 1);
+      detectModelChanges(fixture);
+
+      expect(inputEl.classList).toContain('ng-dirty');
```

Comment:
It should actually not mark the control dirty based on programmatic changes. Dirtiness is a reflection of UI changes only (what the user does). We're probably calling the change callback somewhere we shouldn't be.